cdp/smart_contract.py

```diff
@@ -131,6 +136,16 @@ def contract_address(self) -> str:
         """
         return self._model.contract_address
 
+    @property
+    def contract_name(self) -> str:
+        """Get the contract address of the smart contract.
+
+        Returns:
+            The contract address.
+
+        """
+        return self._model.contract_name
+
     @property
     def deployer_address(self) -> str:
```

Comment:
I think now a lot of these other properties will be only set if is_external is false.

This is the NodeJS change we had to make to support this:
https://github.com/coinbase/coinbase-sdk-nodejs/commit/94680a69f43473f3e08233e6da6c04974e8d4cbc#diff-2c6bead48767df9b8a842ac4887f9e676b0b38a3b6fc2c4bdb7ff6e2d9ddbb7dR227

Reply:
I added a check of the external fields before returning. If it looks good I'll apply it to the nodejs change

Reply:
> If it looks good I'll apply it to the nodejs change

I just linked the change that was made to the NodeJS to handle this!

Reply:
We'll also want to handle #sign, #wait, #broadcast, etc... to ensure they throw appropriate errors when called on external contracts

Reply:
Gotcha, updated! Thank you